src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -249,7 +249,7 @@ public TypeSymbolWithAnnotations AsNullableReferenceOrValueType(CSharpCompilatio
             // In this case we delay asking this question as long as possible.
             if (typeSymbol.TypeKind != TypeKind.TypeParameter)
             {
-                if (typeSymbol.IsReferenceType)
+                if (!typeSymbol.IsValueType)
```

Comment:
One of several instances where we checked `IsReferenceType` rather than `!IsValueType`. (Checking `IsReferenceType` misses unconstrained type parameters.)